packages/core-data/src/actions.js

```diff
@@ -534,6 +404,32 @@ export const saveEntityRecord = (
 	}
 };
 
+function evaluateOptimizedEdits( kind, name, recordId, record ) {
+	return async ( { select, dispatch } ) => {
+		// Evaluate optimized edits.
+		// (Function edits that should be evaluated on save to avoid expensive computations on every edit.)
+		for ( const [ key, value ] of Object.entries( record ) ) {
+			if ( typeof value === 'function' ) {
+				const evaluatedValue = value(
+					select.getEditedEntityRecord( kind, name, recordId )
+				);
+
+				await dispatch.editEntityRecord(
+					kind,
+					name,
+					recordId,
+					{
+						[ key ]: evaluatedValue,
+					},
+					{ undoIgnore: true }
+				);
```

Comment:
There's one thing I don't understand about this evaluation code: when I perform `saveEntityRecord` where there are both direct and lazy edits:
```js
saveEntityRecord( kind, name, {
  title: ( oldTitle ) => oldTitle + ' (new)',
  content: 'Hello',
}
```
Then `editEntityRecord` will be called with the new `title`, but not with `content`. Why is the `edits` state updated in such seemingly arbitrary way?